pkg/ddevapp/utils.go

```diff
@@ -483,3 +483,33 @@ func (app *DdevApp) GetRelativeWorkingDirectory() string {
 	pwd, _ := os.Getwd()
 	return app.GetRelativeDirectory(pwd)
 }
+
+// HasCustomCert returns true if the project uses a custom certificate
+func (app *DdevApp) HasCustomCert() bool {
+	customCertsPath := app.GetConfigPath("custom_certs")
+	certFileName := fmt.Sprintf("%s.crt", app.Name)
+	if !globalconfig.DdevGlobalConfig.IsTraefikRouter() {
+		certFileName = fmt.Sprintf("%s.crt", app.GetHostname())
+	}
+	return fileutil.FileExists(filepath.Join(customCertsPath, certFileName))
+}
+
+// CanUseHTTPOnly returns true if the project can be accessed via http only
+func (app *DdevApp) CanUseHTTPOnly() bool {
+	switch {
+	// Gitpod and Codespaces have their own router with TLS termination
+	case nodeps.IsGitpod() || nodeps.IsCodespaces():
+		return true
```

Comment:
So the original condition was wrong?

Because before refactoring it was:
```go
if !nodeps.IsGitpod() && !nodeps.IsCodespaces()
```

And after:
```go
if nodeps.IsGitpod() || nodeps.IsCodespaces()
```

But it should be with inverse to make the same checks as before:
```go
if !(nodeps.IsGitpod() || nodeps.IsCodespaces())
```

Reply:
Maybe I'm confused, but it can easily be tested. We should *not* be using https on gitpod/codespaces, so the return value should be true here. 

The logic was the opposite before this PR I think?

Reply:
Easy to test on gitpod by clicking the button on HEAD

Reply:
I thought we needed a Docker push for Gitpod (after each release) to update it.

Reply:
Yes, the logic has become the opposite. This is what caught my eye.

Reply:
> I thought we needed a Docker push for Gitpod (after each release) to update it.

That's for ordinary users of ddev-gitpod-launcher. 

For DDEV HEAD, just https://gitpod.io/?autostart=true#https://github.com/ddev/ddev

Reply:
Okay, I see that Gitpod is working, thanks for the info.

Reply:
Thanks for checking this so carefully and paying attention. I'll try to take a look as well.